lib/vtls/mbedtls.c

```diff
@@ -924,13 +961,14 @@ static void mbedtls_close(struct Curl_cfilter *cf, struct Curl_easy *data)
   struct ssl_connect_data *connssl = cf->ctx;
   struct ssl_backend_data *backend = connssl->backend;
   char buf[32];
-  (void) data;
 
   DEBUGASSERT(backend);
 
   /* Maybe the server has already sent a close notify alert.
      Read it to avoid an RST on the TCP connection. */
+  backend->io_data = data;
   (void)mbedtls_ssl_read(&backend->ssl, (unsigned char *)buf, sizeof(buf));
+  backend->io_data = NULL;
```

Comment:
Though I'm loathe to bring up conn->data (which was eliminated about 2 years ago) is there a better way then the association of data like this at a bunch of seemingly arbitrary points? Yes I'm sure they're not arbitrary but I'm struggling to follow how this works. Is there a more central location where data could be assigned, like vtls for example?

Reply:
I agree that this is looking for a more elegant solution. The sprinkling of those assignments in the code is not to my liking really, either.